docs/architecture.md

```diff
@@ -79,4 +79,255 @@ When building new client features, there are a few things worth noting about the
+
+##### Unit Testing
+
+Unit tests cover util functions and other non React based functions. These tests are stored  in the `/unit `folder and can be organized into folders depending on their parent page and/or screen. Use Jest to test each function using `describe` to group together the utils file and the specific function. Then each test is described using `it`. Some functions are very basic and don't need a test. Use your best judgment if a test is needed.
+
+_Example_
+``` ts
+describe('Project view screen utils', () => {
+  describe('getDisplayNameFromK8sResource', () => {
+    it('returns the display name from resource metadata annotations', () => {
+      // Test 1
+    });
+    it('returns the resource name if display name is not present', () => {
+      // Test 2
+    });
+  });
+  // Write similar test cases for other functions in the utils file
+  describe('getSecretDescription', () => {
+    // Test case for getSecretDescription function
+  });
+});
+```
+
+##### Integration Testing
+
+Integration tests will be conducted on specific screens in the app using storybook stories and the playwright testing framework.
+
+
+###### Test setup steps
+
+1. Identify view components that don't require any props and create a story around the entire screen component that handles the business logic.
+2. Create a default export story configuration for the selected component and its child stories. Specify the API mocks using the `msw` parameter.
+
+Configure API mocks using Mock Service Worker (MSW) to simulate API calls. Only mock GET requests for fetching k8 resources; do not mock POST, PUT, or DELETE calls as the tests are run locally.
+
+
+``` ts
+export default {
+  component: ProjectView,
+  parameters: {
+    msw: {
+      handlers: [
+        rest.get(
+       '/api/k8s/apis/route.openshift.io/v1/namespaces/test-project/routes/test-notebook',
+          (req, res, ctx) => res(ctx.json(mockRouteK8sResource({}))),
+        ),
+      ],
+    },
+  },
+} as Meta<typeof ProjectView>;
+```
+
+
+3. Each exported member of the file (except the default export) represents a story and can have various parameters. For example, the a11y parameter can be used to set a custom root element for accessibility tests, such as modals. Optionally setup each test by defining the `play` function,  such as opening a modal, to ensure correct accessibility testing.
+
+``` tsx
+export const EditModel = {
+  render: Template,
+  parameters: {
+    a11y: {
+      // need to select modal as root
+      element: '.pf-c-backdrop',
+    },
+  },
+  play: async ({ canvasElement }) => {
+    // load page and wait until settled
+    const canvas = within(canvasElement);
+    await canvas.findByText('Test Inference Service', undefined, { timeout: 5000 });
+
+    // user flow for editing a project
+    await userEvent.click(canvas.getByLabelText('Actions', { selector: 'button' }));
+    await userEvent.click(canvas.getByText('Edit', { selector: 'button' }));
+  },
+};
+```
```

Comment:
Can we add a link to the docs(https://playwright.dev/docs/locators)? There are other locators apart from `Label` but we have only provided examples for `getByLabel`, `getByText`, and `getByRole` throughout this doc. The user might have to choose between these and the other locators.

Reply:
+1 I like this idea, great suggestion!